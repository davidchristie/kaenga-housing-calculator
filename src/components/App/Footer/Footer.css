.Footer li {
  display: inline;
  padding: 10px;
}

.Footer ul {
  list-style: none;
  margin: 0;
  padding: 0;
}
