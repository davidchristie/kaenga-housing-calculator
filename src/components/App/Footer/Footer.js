import React, { Component } from 'react'
import { Col, Grid, Panel } from 'react-bootstrap'

class Footer extends Component {
  render () {
    return (
      <Grid>
        <Col md={2} />
        <Col md={8}>
          <Panel style={{textAlign: 'center'}}>
            <p>
              <strong>
                Images &copy; Noah Fagan
              </strong>
            </p>
          </Panel>
        </Col>
        <Col md={2} />
      </Grid>
    )
  }
}

export default Footer
