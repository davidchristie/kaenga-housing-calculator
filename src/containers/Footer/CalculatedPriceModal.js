import React, { Component } from 'react'
import { Button, Modal } from 'react-bootstrap'
import { connect } from 'react-redux'

import hideCalculatedPriceModal from '../../actions/hideCalculatedPriceModal'

class CalculatedPriceModal extends Component {
  render () {
    return (
      <Modal show={this.props.show} onHide={this.props.toggle}>
        <Modal.Header closeButton />
        <Modal.Body>
          <p>
            Great, weâ€™ve calculated an estimated price. This will update as you tell us more.
          </p>
        </Modal.Body>
        <Modal.Footer>
          <Button
            bsStyle='primary'
            onClick={this.props.hide}
          >
            Close
          </Button>
        </Modal.Footer>
      </Modal>
    )
  }
}

const withState = connect(
  state => ({
    show: state.showCalculatedPriceModal
  }),
  dispatch => ({
    hide: () => dispatch(hideCalculatedPriceModal())
  })
)

export default withState(CalculatedPriceModal)
