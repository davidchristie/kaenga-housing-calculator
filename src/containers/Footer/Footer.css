.Footer {
  background: #9e0914;
  color: white;
  padding: 10px;
}
