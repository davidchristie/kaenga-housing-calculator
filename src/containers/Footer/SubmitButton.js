import React, { Component } from 'react'
import { Button } from 'react-bootstrap'
import { connect } from 'react-redux'

import { postRegistration } from '../../actions/register'

class SubmitButton extends Component {
  render () {
    return (
      <Button onClick={this.props.postRegistration}>
        Submit
      </Button>
    )
  }
}

export default connect(
  state => {
    return {}
  },
  dispatch => {
    return {
      postRegistration: () => dispatch(postRegistration())
    }
  }
)(SubmitButton)
