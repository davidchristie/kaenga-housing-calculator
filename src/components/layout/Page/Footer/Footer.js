import React, { Component } from 'react'
import { Well } from 'react-bootstrap'
import { Link } from 'react-router'

import './Footer.css'

class Footer extends Component {
  render () {
    return (
      <Well className='Footer' style={{textAlign: 'center'}}>
        <ul>
          <li>
            <strong>
              &copy; 2017 Kaenga
            </strong>
          </li>
          <li>
            <Link to='/privacy'>Privacy</Link>
          </li>
        </ul>
      </Well>
    )
  }
}

export default Footer
