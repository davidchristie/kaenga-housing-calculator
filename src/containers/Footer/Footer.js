import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Nav, Navbar } from 'react-bootstrap'
import { connect } from 'react-redux'

import { getSelectedSuburbs } from '../../selectors/suburbs'
import PriceRange from '../reports/PriceRange'
import CalculatedPriceModal from './CalculatedPriceModal'
import './Footer.css'
import SubmitButton from './SubmitButton'

class Footer extends Component {
  render () {
    if (this.props.show) {
      return (
        <Navbar className='Footer' fixedBottom>
          <span style={{fontSize: '1.4em'}}>
            Cost: <PriceRange />
          </span>
          <Nav pullRight>
            <SubmitButton />
          </Nav>
          <CalculatedPriceModal />
        </Navbar>
      )
    } else {
      return <span />
    }
  }
}

Footer.propTypes = {
  show: PropTypes.bool.isRequired
}

function isValidDesign (state) {
  return !(state.form && state.form.design && state.form.design.syncErrors)
}

function isValidLocation (state) {
  return getSelectedSuburbs(state).length > 0
}

export default connect(
  state => {
    return {
      show: isValidDesign(state) && isValidLocation(state)
    }
  }
)(Footer)
