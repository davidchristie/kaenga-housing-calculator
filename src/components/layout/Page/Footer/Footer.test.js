import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import Footer from './Footer'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Footer />)).toMatchSnapshot()
})
