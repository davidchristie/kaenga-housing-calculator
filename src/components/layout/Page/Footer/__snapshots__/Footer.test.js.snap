exports[`test matches snapshot 1`] = `
<Grid
  bsClass="container"
  className="Footer"
  componentClass="div"
  fluid={false}>
  <Col
    bsClass="col"
    componentClass="div"
    md={2} />
  <Col
    bsClass="col"
    componentClass="div"
    md={8}>
    <Panel
      bsClass="panel"
      bsStyle="default"
      defaultExpanded={false}
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <ul>
        <li>
          <Unknown
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/privacy">
            Privacy
          </Unknown>
        </li>
      </ul>
      <strong>
        © 2017 Kaenga
      </strong>
      <br />
      <strong>
        Images © Noah Fagan
      </strong>
    </Panel>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={2} />
</Grid>
`;
